lib/OperatingSystems/DefaultOs.php

```diff
@@ -0,0 +1,168 @@
+<?php
+/**
+ * @author Frank Karlitschek <frank@nextcloud.com>
+ *
+ * @license AGPL-3.0
+ *
+ * This code is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License, version 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License, version 3,
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>
+ *
+ */
+
+namespace OCA\Serverinfo\OperatingSystems;
+
+
+/**
+ * Class Ubuntu
+ *
+ * @package OCA\ServerInfo\OperatingSystems
+ */
+class DefaultOs {
+
+	/**
+	 */
+	public function __construct() {
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function supported() {
+		return true;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getHostname() {
+		$hostname = shell_exec('hostname');
+		return $hostname;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getMemory() {
+		$memory = shell_exec('cat /proc/meminfo  | grep -i \'MemTotal\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		$memory = explode(' ',$memory);
+		$memory = round($memory[0]/1024);
+		if ($memory<1024) {
+			$memory = $memory.' MB';
+		} else {
+			$memory = round($memory/1024,1).' GB';
+		}
+		return $memory;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getCPUName() {
+		$cpu   = shell_exec('cat /proc/cpuinfo  | grep -i \'Model name\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		$cores = shell_exec('cat /proc/cpuinfo  | grep -i \'cpu cores\' | cut -f 2 -d ":" | awk \'{$1=$1}1\'');
+		if ($cores == 1) $cores = ' ('.$cores.' core)'; else $cores = ' ('.$cores.' cores)';
+		return $cpu.' '.$cores;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getTime() {
+		$uptime = shell_exec('date');
+		return $uptime;
+	}
+
+	/**
+	 * @return string
+	 */
+	public function getUptime() {
+		$uptime = shell_exec('uptime -p');
+		return $uptime;
+	}
+
+
+	/**
+	* @return string
+	*/
+	public function getTimeServers() {
+		$uptime = shell_exec('cat /etc/ntp.conf |grep  \'^pool\' | cut -f 2 -d " "');
```

Comment:
I see this just a bit problematic for systems using systemd, like major server distributions Debian and Ubuntu. In this cases the NTP servers would be in `/etc/systemd/timesyncd.conf` instead. So the question is how to handle that cases - simply using systemd-based checks here, or creating own `OperatingSystem` classes for systemd-systems and non-systemd-systems?

Reply:
Yes. Unfortunately there are a lot more differences between different operating systems. There is a system in place to have different backends for the different distributions. At the moment this has only the default backend which should often works. Tested with Ubuntu here. But this requires more backends for different distros. As you described.

Reply:
> Tested with Ubuntu here.

Which one? Because systemd is already in place since quite some time and seems to be the de-facto standard, that has the most coverage easily.

Reply:
Exactly. As I remember systemd is in place since Ubuntu 16.04 (first release on April 21, 2016) and Debian 8 (first release on April 26, 2015). `FreeBSD` and sort of needs a special handling anyway - but that's another topic for later.

Reply:
Great feedback. I'm reading this file now additionally. On my Ubuntu 18.10 the config is in /conf/ntp.conf Probably because this is an older install and the config is migrated from an old Ubuntu